src/praisonai/praisonai/tool_resolver.py

```
@@ -486,6 +486,62 @@ def get_local_tool_classes(self) -> Dict[str, Any]:
+        from pathlib import Path
+        from ._safe_loader import load_user_module
+        
+        classes: Dict[str, Any] = {}
+        try:
+            for py_file in Path(tools_dir).glob("*.py"):
+                if py_file.name.startswith("__"):
+                    continue
+                module = load_user_module(py_file, name=f"tools_{py_file.stem}")
+                if module is not None:
+                    classes.update(self._extract_tool_classes(module))   # reuse resolver's own predicate
+        except Exception as e:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)